services/src/main/java/io/scalecube/services/ServiceScanner.java

```diff
@@ -21,8 +24,16 @@ private ServiceScanner() {
    */
   public static List<ServiceRegistration> scanServiceInfo(ServiceInfo serviceInfo) {
     return Arrays.stream(serviceInfo.serviceInstance().getClass().getInterfaces())
-        .map(serviceInterface -> new InterfaceInfo(serviceInterface, serviceInfo.tags()))
-        .filter(interfaceInfo -> interfaceInfo.serviceInterface.isAnnotationPresent(Service.class))
+        .filter(serviceInterface -> serviceInterface.isAnnotationPresent(Service.class))
+        .map(
+            serviceInterface -> {
+              Map<String, String> userTags = serviceInfo.tags();
+              Map<String, String> apiTags = Reflect.serviceTags(serviceInterface);
+              Map<String, String> buffer = new HashMap<>(apiTags);
+              // user tags override tags from @Service
+              buffer.putAll(userTags);
```

Comment:
Looks like you doing this correctly, u override annotation tags with programmatic tags. Also look at this `io.scalecube.services.ServiceReference#mergeTags`  for details. This is FYI.

Reply:
i have seen